### PR TITLE
(5.x) Merge 4.x

### PR DESCRIPTION
**Pull request by asmorkalov**
Contrib: https://github.com/opencv/opencv_contrib/pull/3688
Extra: https://github.com/opencv/opencv_extra/pull/1159

#24724 from tomoaki0705:carotene_warnings
#25120 from LaurentBerger:I25103
#25122 from unnonouno:pqueue
#25123 from klatism:zlib-upgrade
#25130 from CSBVision:patch-6
#25141 from Kumataro:fix25140_4.x
#25143 from Aman-Vishwakarma1729:4.x
#25162 from mshabunin:doc-upgrade-fix
#25163 from dkurt:onnx_graph_ref

Previous "Merge 4.x": #25119

### CODE REVIEW

File: modules/features2d/include/opencv2/features2d.hpp
Comment (asmorkalov) [inline modules/features2d/include/opencv2/features2d.hpp:1583]: Merge mistake.

File: modules/dnn/test/test_layers.cpp
Comment (asmorkalov) [inline modules/dnn/test/test_layers.cpp:1806]: @fengyuentau @vpisarev @dkurt I have to add `.t()` to have the same shapes. Is it right solution?

Comment (fengyuentau) [reply modules/dnn/test/test_layers.cpp:1806]: I guess it relates to 1d mat changes. `resAxesn` is 1d tensor with shape [n], when it is loaded into a Mat, if I remember correctly, it was of shape [n, 1] before and now [1, n] after 1d mat is supported on 5.x.

**Comment by mshabunin** [comment]
@asmorkalov , can we also add #25171 (documentation fix) to this merge train if that PR is merged today? It should fix doc warnings on BuildBot.

File: modules/dnn/test/test_layers.cpp
Comment (asmorkalov) [reply modules/dnn/test/test_layers.cpp]: Reshape does not support 1d for now. I filed a bug about it. Also I re-implemented check with 2d case without `.t()` call.

**Comment by asmorkalov** [comment]
@mshabunin it's hard to add yet another commit to the merge right now, too many conflicts. I'll repeat the process again as soon as it's merged.

File: modules/dnn/test/test_layers.cpp
Comment (dkurt) [reply modules/dnn/test/test_layers.cpp]: `Mat a = output.reshape(output.total(), 1);` also a valid solution

Comment (asmorkalov) [reply modules/dnn/test/test_layers.cpp]: @fengyuentau @dkurt I reworked it a bit. Could you look again?

Comment (fengyuentau) [reply modules/dnn/test/test_layers.cpp]: Forget my previous comment since it is kind of misleading.

1. Shapes of result and reference are checked in 1803.
2. So lateral is checking values. `Mat a` is still 2d, e.g. [n, 1]. However, loading a vector into Mat using `Mat(resReduceMean[i])`, we now get a Mat of shape [n]. I guess `.t()` creates the extra appended dimension.

Sugguestions to change:

1. Use other asserting function to check value which does not check shape. Or,
2. Drop shape checking at line 1803 and reshaping at line 1805. Initialize the reference Mat with correct shape and compare with output directly:

```.diff
diff --git a/modules/dnn/test/test_layers.cpp b/modules/dnn/test/test_layers.cpp
index e9e93a7853..587c2275a6 100644
--- a/modules/dnn/test/test_layers.cpp
+++ b/modules/dnn/test/test_layers.cpp
@@ -1802,10 +1802,7 @@ TEST(Layer_Test_ReduceMean, accuracy_input_0)
 
         EXPECT_EQ(gt_shape, shape(output));
 
-        // reshape does not support 1d for now: https://github.com/opencv/opencv/issues/25174
-        Mat a = output.reshape(1, output.total());
-        Mat gt = Mat(resReduceMean[i].size(), 1, CV_32FC1, &resReduceMean[i][0]);
-        normAssert(gt, a);
+        normAssert(output, Mat(gt_shape, CV_32F, resReduceMean[i].data()));
     }
 }
```

**Comment by vpisarev** [comment]
@asmorkalov, once this PR is repaired, can we squeeze https://github.com/opencv/opencv/pull/25217 in?

**Comment by asmorkalov** [comment]
@dkurt @opencv-alalek @fengyuentau The PR ready for review. Let's merge it ASAP and I'll do another iteration with required changes.

File: modules/dnn/test/test_layers.cpp
Comment (asmorkalov) [reply modules/dnn/test/test_layers.cpp]: Done!

**Review by fengyuentau — APPROVED** [review]
dnn part looks good 👍

**Pull request closed without merging**
